### ISSUE TITLE
Accelerated conv backward is gated off for every strided and every grouped convolution

### ISSUE THREAD

**Issue by kafonek** [kafonek/burn#7]
The accelerated implicit-GEMM convolution backward is gated off for **every strided convolution and every grouped (depthwise) convolution**. Both gradients fall back to slower paths, and the data-gradient fallback additionally performs a layout round trip whose inner conversions cancel.

This matters for how the published convolution-backward numbers should be read: measurements taken at `stride=1, groups=1` describe the *accelerated* path. A downsampling or depthwise network never reaches it.

### The gates

`burn-cubecl 0.22.0-pre.2`.

Data gradient — `src/kernel/conv/backward_data/implicit_gemm/launch.rs:94`:

```rust
if options.groups != 1 || options.stride.iter().any(|&s| s != 1) {
    return Err(ConvSetupError::Groups(options.groups));
}
```

Weight gradient — `src/kernel/conv/backward_weight/implicit_gemm/launch.rs:94`:

```rust
if options.groups != 1 {
    return Err(ConvSetupError::Groups(options.groups));
}
```

So:

| Convolution | Data gradient | Weight gradient |
|---|---|---|
| `stride=1, groups=1` | accelerated | accelerated |
| `stride>1, groups=1` | **fallback** | accelerated |
| any `groups>1` (incl. depthwise) | **fallback** | **fallback** |

The forward has no equivalent restriction, so a strided layer can run an accelerated forward and a fallback data gradient. The weight-gradient restriction is groups-only, which suggests the stride gate on the data gradient is an unfinished edge case in the im2col reader rather than a fundamental limit — worth confirming with whoever wrote it.

### The data-gradient fallback permutes in a circle

`burn-cubecl` stores NCHW and the kernels want NHWC, so each operation converts at its own boundary. On the fallback path the conversions cancel:

1. `conv_data_backward` converts NCHW→NHWC (`src/kernel/conv/base.rs:163-165`)
2. `conv_data_backward_fallback` immediately converts NHWC→NCHW (`src/kernel/conv/backward_data/fallback.rs:42-43`), with the in-tree comment that there are no NHWC kernels for `conv_transpose`
3. `conv_transpose2d` runs
4. the fallback converts back (`fallback.rs:89`)
5. `conv_data_backward` converts back again (`base.rs:188`)

Four of those five conversions are round trips. This fires on every strided and every grouped convolution — that is, on most downsampling layers of a real vision network.

Separately and unconditionally: a backward pass pays six layout conversions to the forward's three, and the output gradient is converted twice, because the data gradient and the weight gradient are independent backend operations (`burn-autodiff-0.22.0-pre.2/src/ops/module.rs:485` and `:495`) that do not share the converted tensor.

### Why this is worth acting on

A public report exists of a small object-detection network training far slower than its PyTorch equivalent, with a per-phase breakdown showing the backward pass at roughly 4.6x the forward while the PyTorch equivalent sits below 1x. That network is depthwise and strided. Everything above says its convolution backward never touched the accelerated kernels at all, so a reader who benchmarks `stride=1, groups=1` and concludes the accelerated path is merely slow will be measuring a path that model does not execute.

For calibration, here is the accelerated path on 1x NVIDIA H100 80GB HBM3, `burn 0.22.0-pre.2` / `cubecl 0.11.0-pre.2` / `cubek 0.3.0-pre.2`, Linux container, 20 warmup and 50 timed rounds, 3x3 kernel, padding `Same`, `stride=1`, `groups=1`:

| shape | dtype | forward | data gradient | weight gradient |
|---|---|---|---|---|
| `n256 c64 32x32 k64` | f32 | 0.339 ms | 0.767 ms — 25.2 TFLOP/s | 7.610 ms — 2.54 TFLOP/s |
| `n256 c64 32x32 k64` | bf16 | 0.301 ms | 0.441 ms — 43.9 TFLOP/s | 7.052 ms — 2.74 TFLOP/s |
| `n32 c256 64x64 k256` | f32 | 2.150 ms — 71.9 TFLOP/s | 3.928 ms — 39.4 TFLOP/s | 5.901 ms — 26.2 TFLOP/s |
| `n32 c256 64x64 k256` | bf16 | 1.213 ms — 127.4 TFLOP/s | 2.435 ms — 63.5 TFLOP/s | 6.204 ms — 24.9 TFLOP/s |

The weight gradient dominates the accelerated backward; that has its own root cause and is written up separately (this fork's `cubecl` repository, the convolution backward issue). The point here is that these are the **best-case** numbers.

### How to attribute a slow training step to a specific gradient

This does not need a profiler and it works on any model:

- **Weight gradient, timed exactly.** `conv2d_weight_backward` is public backend API (`burn-tensor-0.22.0-pre.2/src/tensor/module.rs:614`). Call it directly. No autodiff graph, no data gradient mixed in.
- **Data gradient, timed by subtraction.** Run autodiff backward with the input tracked and the weight **not** tracked, then subtract a forward timed the same way. Burn skips `conv2d_weight_backward` entirely when the weight is untracked — it is guarded by `if let Some(node) = node_weight` (`burn-autodiff-0.22.0-pre.2/src/ops/module.rs:532-534`) — so this isolates the data gradient.

### Suggested fixes, cheapest first

1. **Skip the cancelling conversions.** Either give `conv_transpose2d` an NHWC entry point, or have `conv_data_backward` not convert when it knows it is dispatching to the fallback. Self-contained, easy to review, independent of any performance work.
2. **Lift the `stride != 1` gate on the accelerated data gradient.** The weight gradient has no such gate, so the restriction is data-gradient-specific. Measure the cliff first.
3. **Support `groups > 1` on the accelerated path**, or document clearly that depthwise convolutions have no accelerated backward. Depthwise is the default building block of most efficient vision backbones, so silently falling back is expensive and invisible.

### Not verified

- **I did not measure a strided or a depthwise convolution.** The gates are read from source; the size of the resulting cliff is unmeasured. Adding a `stride=2` shape and a `groups=c` shape to the benchmark above is a small change and is the right next step.
- I did not read `conv_transpose2d`'s own kernel quality, which sets the floor for how much fix 1 and fix 2 can win.
- I did not check whether the `groups` restriction is a limitation of the im2col reader or of the matmul components below it.
- I did not check the corresponding gates on any backend other than the CUDA one.
